classes-recipe/platsch.bbclass

```diff
@@ -0,0 +1,138 @@
+#
+# Description:
+#
+# "platsch" expects raw bitmaps in a format that the target's graphics
+# hardware's display controller is able to scan out without further pre-
+# processing. This class helps creating them starting from conventionally
+# formatted images whose filenames have to satisfy the following scheme:
+#
+#   <base>-<width>x<height>.<extension>
+#
+# In the recipe the conventional images to be converted are listed and the raw
+# format that they shall be converted to is specified:
+#
+#   SPLASH_IMAGES = "splash-1280x800.png splash-800x480.png"
+#   SPLASH_FORMAT = "RGB565"
+#
+# Of course, those source images have to be listed in SRC_URI beforehand.
+#
+# If the raw format that you need for your hardware is not supported by this
+# class up to now, you are invited to create a corresponding pull request.
+#
+# ImageMagick's BMP encoder always creates "bottom up" bitmap data whereas the
+# i.MX6/i.MX8 expect the pixel array to be stored "top down" so that one needs
+# to compensate for it by flipping the image during the conversion (see [1] and
+# [2] further down). To make things like that possible there is the bitbake
+# parameter MGK_CUSTOM_OPTION which is weakly predefined to flip the image and
+# which can be overwritten in any recipe utilizing this class if required.
+#
+# If the resulting images shall not be installed to the default location (which
+# is /usr/share/platsch) choose another directory via
+#
+#   SPLASH_DATADIR = "/some/different/directory"
+#
+# Finally the class automatically appends the converted images to the
+# FILES:${PN} parameter.
+#
+
+SPLASH_IMAGES ??= ""
+SPLASH_IMAGES[doc] = "Space separated list of conventionally formatted images (*.png, *.jpg etc.) to be converted by this class."
+
+SPLASH_FORMAT ??= "undefined"
+SPLASH_FORMAT[doc] = "Target format that the graphics hardware's display controller is able to scan out."
+
+MGK_CUSTOM_OPTION ??= "-flip"
+MGK_CUSTOM_OPTION[doc] = "Custom options to be injected into ImageMagick's commandline, predefined for i.MX hardware."
+
+SPLASH_DATADIR ??= "${datadir}/platsch"
+SPLASH_DATADIR[doc] = "Directory where the converted images will be installed."
+
+DEPENDS = "imagemagick-native"
+
+# Remove and re-create ${B} so that it is guaranteed to be empty as otherwise
+# the glob expression in do_install() would probably install out-of-date
+# leftovers from older runs of the recipe using this class.
+do_compile[cleandirs] = "${B}"
+
+platsch_do_compile() {
+    for IMG in ${SPLASH_IMAGES}; do
+        # filename has to be <base>-<width>x<height>.<extension>
+        echo $IMG | sed -E -e "s/(.*)-(.*)x(.*)\.(.*)/\1 \2 \3 \4/" |
+        {
+            read BASE WIDTH HEIGHT EXT
+            if [ -z "${BASE}" -o -z "${WIDTH}" -o -z "${HEIGHT}" -o -z "${EXT}" ]; then
+                bbfatal "platsch.bbclass: filename \"${IMG}\" does not satisfy scheme <basename>-<width>x<height>.<extension>: BASE=\"${BASE}\", WIDTH=\"${WIDTH}\", HEIGHT=\"${HEIGHT}\", EXT=\"${EXT}\"."
```

Comment:
```suggestion
                bbfatal "platsch.bbclass: filename \"${IMG}\" does not satisfy scheme <base>-<width>x<height>.<extension>: BASE=\"${BASE}\", WIDTH=\"${WIDTH}\", HEIGHT=\"${HEIGHT}\", EXT=\"${EXT}\"."
```

Reply:
I fear that I am finally routine-blinded... thanks for stumbling across it!